de/graph.py

```diff
@@ -9,18 +9,24 @@ def load_w():
     Loads w from csv file and returns dataframe with gene symbols attached to w values.
     """
     path_here = dirname(dirname(__file__))
-    w = pd.read_csv(join(path_here, "de/data/w.csv"), header=None)
+    w = pd.read_csv(join(path_here, "de/data/w_new.csv"), header=None)
     genes = np.loadtxt(join(path_here, "de/data/node_Index.csv"), dtype=str)
     w.columns = genes
     w.index = genes
     return w
 
-def remove_POLR2A(w):
+def remove(w):
     """
-    Removes POLR2A from w matrix.
+    Removes POLR2A and genes whose expression level equals zero under control condition from w matrix.
     """
     w = w.drop(["POLR2A"], axis=0)
     w = w.drop(["POLR2A"], axis=1)
+    m = w.columns
+    w_2 = w.to_numpy()
+    for i in range(len(m)):
+        if w_2[0, i] == 0:
```

Comment:
Why do you have to remove those genes with zero expression?

Reply:
There are some all-zeros columns in w_new matrix after normalizing. Since the adjacency matrix to be calculated in the PageRank function need to be square and the sum of each column should be one, I removed those genes from the matrix.